open-hackathon-client/src/client/static/js/views/oh.manage.notice.js

```diff
@@ -0,0 +1,191 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+(function($, oh) {
+    var currentHackathon = oh.comm.getCurrentHackathon();
+
+    function getNoticeDescription(items) {
+        var noticeDescriptionByType = {
+            0: '黑客松信息',
+            1: '用户信息',
+            2: '实验信息',
+            3: '获奖信息',
+            4: '模板信息'
+        }
+
+        var length = items.length;
+        for(var i = 0; i < length; ++i) {
+            items[i].type = noticeDescriptionByType[items[i].type];
+        }
+    }
+
+    function toggleTable(status){
+        if(status == 0){
+            $('#hackathon_notice_list_table').show();
+            $('#hackathon_notice_add_table').hide();
+        }else{
+            $('#hackathon_notice_list_table').hide();
+            $('#hackathon_notice_add_table').show();
+        }
+    }
+
+    // call api to get hackathon notices
+    function getHackathonNotices(){
+        oh.api.hackathon.notice.list.get({
+            query: {
+                hackathon_name: currentHackathon,
+                order_by: 'time'
+            }
+        }, function(data) {
+            if(data.error){
+                alert(data.error.message);
+            }else{
+                getNoticeDescription(data.items);
+                $('#hackathon_notice_list').empty().append($('#hackathon_notice_list_template').tmpl(data.items));
+                oh.comm.removeLoading();
+            }
+        });
+    }
+
+    // call api to add hackathon notice
+    function addHackathonNotice(){
+        data = {};
+        data.content = $.trim($('#add_notice_content').val());
+        data.link    = $.trim($('#add_notice_link').val());
+
+        return oh.api.admin.hackathon.notice.post({
+            body: data,
+            header: {
+                hackathon_name: currentHackathon
+            }
+        }, function(data) {
+            if(data.error){
+                alert(data.error.message);
```

Comment:
Please don't use alert which is usually blocked by browsers. Try oh.comm.alert(name may vary)